src/tensorflow.cc

```diff
@@ -1332,62 +1342,196 @@ AutoCompleteHelper::FixBatchingSupport()
+
+  bool is_empty_io = IsEmptyIO(io_config);
+  if (!is_empty_io && !io_config.Find("name", &tmp)) {
+    return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INVALID_ARG,
+          (std::string("'name' is a required field for all inputs/outputs."))
+              .c_str());
+  }
+  
+  // If this I/O is empty then we are using the model
+  // to autocomplete.
+  if (is_empty_io) {
+    io_config.SetStringObject("name", io->name_);
+  }
+  
+  bool found_config_data_type = io_config.Find("data_type", &tmp);
+  std::string data_type_str;
+  tmp.AsString(&data_type_str);
+  bool should_auto_complete_data_type =
+      !found_config_data_type || DataTypeIsInvalid(data_type_str);
+  if (should_auto_complete_data_type) {
+    io_config.SetStringObject("data_type", ConvertToModelConfigString(io->data_type_));
+  }
+
+  bool found_dims = io_config.Find("dims", &tmp);
+  bool should_auto_complete_dims = !found_dims || (tmp.ArraySize() == 0);
+  if (should_auto_complete_dims) {
+    RETURN_ERROR_IF_TRUE(
+        io->shape_->rank_ == 0, TRITONSERVER_ERROR_INVALID_ARG,
+        std::string(
+            "Unable to autofill for '" + model_state_->Name() +
+            "': the rank of model tensor '" + io->name_ +
+            "' is 0 and dimensions are not defined"));
+    // The model signature supports batching then the first
+    // dimension is -1 and should not appear in the model
+    // configuration 'dims' that we are creating.
+    triton::common::TritonJson::Value dims(
+        model_state_->ModelConfig(),
+        triton::common::TritonJson::ValueType::ARRAY);
+
+    for (size_t i = (model_support_batching_ ? 1 : 0); i < io->shape_->rank_;
+         ++i) {
+      RETURN_IF_ERROR(dims.AppendInt(io->shape_->dims_[i]));
+    }
+    if (dims.ArraySize() == 0) {
+      RETURN_IF_ERROR(dims.AppendInt(1));
+      triton::common::TritonJson::Value reshape(
+          model_state_->ModelConfig(),
+          triton::common::TritonJson::ValueType::OBJECT);
+      triton::common::TritonJson::Value reshape_dims(
+          model_state_->ModelConfig(),
+          triton::common::TritonJson::ValueType::ARRAY);
+      RETURN_IF_ERROR(reshape.Add("shape", std::move(reshape_dims)));
+      RETURN_IF_ERROR(io_config.Add("reshape", std::move(reshape)));
+    }
+
+    if (found_dims) {
+      RETURN_IF_ERROR(io_config.Remove("dims"));
+    }
+    RETURN_IF_ERROR(io_config.Add("dims", std::move(dims)));
+  } // end of should_auto_complete_dims
+
+  return nullptr;  // Success
+}
+
+void
+AutoCompleteHelper::RemoveFromListByName(
+    const char* name, std::vector<const TRITONTF_IOList*>& list)
+{
+  for (size_t i = 0; i < list.size(); ++i) {
+    const TRITONTF_IOList* current = list.at(i);
+    if (!strcmp(name, current->io_->name_)) {
+      list.erase(list.begin() + i);
+      return;
+    }
+  }
+}
+
+void
+AutoCompleteHelper::CopyList(std::vector<const TRITONTF_IOList*>& dst, const TRITONTF_IOList* src)
```

Comment:
reorder such that it is more like follows:  src, dst

Reply:
Fixed